pkg/restapi/cluster.go

```diff
@@ -108,6 +112,42 @@ func (h clusterHandler) parseCluster(r *http.Request) (*cluster.Cluster, error)
 	return &c, nil
 }
 
+func (h clusterHandler) parseClusterForUpdate(r *http.Request) (*cluster.Cluster, error) {
+	// Read the request body into a buffer
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to read request body")
+	}
+
+	// Decode the request body into a cluster.Cluster object
+	var c cluster.Cluster
+	if err := json.NewDecoder(bytes.NewReader(body)).Decode(&c); err != nil {
+		return nil, errors.Wrapf(err, "unable to decode to cluster.Cluster")
+	}
+
+	// Decode the request body into a models.Cluster object
+	var apiC models.Cluster
+	if err := json.NewDecoder(bytes.NewReader(body)).Decode(&apiC); err != nil {
+		return nil, errors.Wrapf(err, "unable to decode to cluster.Cluster")
+	}
+
+	// Check for nil pointer fields and update if necessary
+	if apiC.ForceTLSDisabled == nil || apiC.ForceNonSslPort == nil {
+		oldCluster, err := h.svc.GetCluster(r.Context(), apiC.ID)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to get cluster {%v}", apiC.ID)
+		}
+		if apiC.ForceTLSDisabled == nil {
+			c.ForceTLSDisabled = oldCluster.ForceTLSDisabled
+		}
+		if apiC.ForceNonSslPort == nil {
+			c.ForceNonSSLPort = oldCluster.ForceNonSSLPort
+		}
+	}
```

Comment:
Could you explain why those flags are handled in a special way?
My understanding is that when we create/update cluster, we require client to send a full new cluster definition to SM, not just the changed fields. We handle it on the `sctool` side by first getting the cluster definition from SM and then applying changes to it:
```go
func (cmd *command) run() error {
	cluster, err := cmd.client.GetCluster(cmd.Context(), cmd.cluster)
	if err != nil {
		return err
	}

	ok := false
	if cmd.Flags().Changed("name") {
		cluster.Name = cmd.name
		ok = true
	}
	if cmd.Flags().Changed("host") {
		cluster.Host = cmd.host
		ok = true
	}
	if cmd.Flags().Changed("port") {
		cluster.Port = cmd.port
		ok = true
	}
```
This way we don't worry that on update not specified value could overwrite previously set value.
If that's the case, I think that those flags should follow the same convention for simplicity.

Reply:
Actually you are right. CLI update gets the cluster first and then applies all what is changed.
There is no need to even keep fields nullable in swagger.

Reply:
done